firedrake/solving_utils.py

```diff
@@ -229,10 +266,38 @@ def split(self, fields):
                         W = V.sub(field_renumbering[index])
                     if cmpt is not None:
                         W = W.sub(cmpt)
-                    bcs.append(type(bc)(W,
-                                        bc.function_arg,
-                                        bc.sub_domain,
-                                        method=bc.method))
+                    if isinstance(bc, DirichletBC):
+                        return DirichletBC(W,
```

Comment:
This breaks user-defined subclasses of `DirichletBC`. TBF, the old code also did. But `DirichletBC` objects have a `reconstruct` method which one should use:
```
return bc.reconstruct(V=W, g=bc.function_arg, sub_domain=bc.sub_domain, method=bc.method)
```

Reply:
Fixed.